subsys/net/l2/ieee802154/ieee802154_mgmt.c

```diff
@@ -37,6 +37,7 @@ enum net_verdict ieee802154_handle_beacon(struct net_if *iface,
 					  uint8_t lqi)
 {
 	struct ieee802154_context *ctx = net_if_l2_data(iface);
+	int header_length;
```

Comment:
my personal style and controversial, therefore strictly optional:

```suggestion
	int beacon_hdr_len;
```

Reasoning (not for you but for certain other reviewers who regularly advocate for C/Linux style var names like `len`or `bhl`):
* "payload len" minus "header len" might cause extra cognitive load as people might understand "MAC payload len minus MHR len" which would be nonsense.
* I don't care about programming C w/o IDE support, I care about readability.